### PR TITLE
feature:Support alarm to WeLink

### PR DESCRIPTION
**Pull request by YunfengGao**
### <Feature description>
- [ ] If this is non-trivial feature, paste the links/URLs to the design doc.
- [x] Update the documentation to include this new feature.
- [x] Tests(including UT, IT, E2E) are added to verify the new feature.
- [x] If it's UI related, attach the screenshots below.
image

### CODE REVIEW

**Comment by wu-sheng** [comment]
@YunfengGao Are you from Huawei team? I think we don't have a committer to verify the WeLink, as none of us is using this.

@WillemJiang and @apache/skywalking-committers Do you know anyone who could help with reviewing codes?

**Comment by YunfengGao** [comment]
> @YunfengGao Are you from Huawei team? I think we don't have a committer to verify the WeLink, as none of us is using this.
> 
> @WillemJiang and @apache/skywalking-committers Do you know anyone who could help with reviewing codes?

Yes, I come from HUAWEI but not WeLink Team. Compared with other IM tools, WeLink has just started and few people use it now. I quite clear with that.
I change my code a little but not run UT again and now I found it failed. I wonder if I can close P.R and roll back my code then submit again ?

**Comment by wu-sheng** [comment]
> I change my code a little but not run UT again and now I found it failed. I wonder if I can close P.R and roll back my code then submit again ?

You could change on your branch directly, or close/polish/submit again. Your choice.
Also, with the project's current status, I still can't find people to review/recheck or test this locally. So, after you finish, I may need to ask around, to request a review from other eyes.

**Comment by WillemJiang** [comment]
> @YunfengGao Are you from Huawei team? I think we don't have a committer to verify the WeLink, as none of us is using this.
> 
> @WillemJiang and @apache/skywalking-committers Do you know anyone who could help with reviewing codes?

There is public version of welink that can be used outside of Huawei. So you don't need ask a Huawei-er to verify it.

**Comment by wu-sheng** [comment]
> There is public version of welink that can be used outside of Huawei. So you don't need ask a Huawei-er to verify it.

Technically, yes, you are correct. But it is really hard to find one. I am asking around.

**Review by wu-sheng — COMMENTED** [review]
The UT could only test webhook is working, but we have no chance to verify whether WeLink actually works.

@kezhenxu94 Any suggestion? Should we merge this blindly?

File: oap-server/server-alarm-plugin/src/test/resources/alarm-settings.yml
Comment (wu-sheng) [inline oap-server/server-alarm-plugin/src/test/resources/alarm-settings.yml:128]: We need to be clear about these ID/secret(s) are dummy, need to change.

File: oap-server/server-alarm-plugin/src/main/java/org/apache/skywalking/oap/server/core/alarm/provider/Rules.java
Comment (wu-sheng) [inline oap-server/server-alarm-plugin/src/main/java/org/apache/skywalking/oap/server/core/alarm/provider/Rules.java:45]:
```suggestion
    private WeLinkSettings welinks;
```

I think this name in codes and config file make more sense.

**Review by wu-sheng — COMMENTED** [review]
@apache/skywalking-committers If anyone wants to and could verify, I am good to wait for. Otherwise, I am going to merge this, as this is harmless even there is a bug.

**Comment by kezhenxu94** [comment]
> The UT could only test webhook is working, but we have no chance to verify whether WeLink actually works.
> 
> @kezhenxu94 Any suggestion? Should we merge this blindly?

When alarm rules are triggered, as long as the webhook request is sent from SkyWalking side I think we’re done, testing whether WeLink works or not after receiving the webhook is beyond our responsibility.

File: oap-server/server-alarm-plugin/src/main/java/org/apache/skywalking/oap/server/core/alarm/provider/welink/WeLinkSettings.java
Comment (kezhenxu94) [inline oap-server/server-alarm-plugin/src/main/java/org/apache/skywalking/oap/server/core/alarm/provider/welink/WeLinkSettings.java:56]: This name looks weird to me, why is “reboot”?

**Comment by wu-sheng** [comment]
> > The UT could only test webhook is working, but we have no chance to verify whether WeLink actually works.
> > @kezhenxu94 Any suggestion? Should we merge this blindly?
> 
> When alarm rules are triggered, as long as the webhook request is sent from SkyWalking side I think we’re done, testing whether WeLink works or not after receiving the webhook is beyond our responsibility.

OK, agree. Let me document here in case bugs showing up in the future.

**SkyWalking community considers this harmless and merges this as a part of the codebase. We have no committer tested. We will not respond to a bug report or enhancement requirement. We could forward this kind of report to @YunfengGao or wait for other volunteers to fix it or support it.**

@apache/skywalking-committers Any objection, please let me know. I will wait 2 days at least.

File: oap-server/server-alarm-plugin/src/main/java/org/apache/skywalking/oap/server/core/alarm/provider/welink/WeLinkSettings.java
Comment (wu-sheng) [reply oap-server/server-alarm-plugin/src/main/java/org/apache/skywalking/oap/server/core/alarm/provider/welink/WeLinkSettings.java:56]: I think he means `robot`?

Comment (YunfengGao) [reply oap-server/server-alarm-plugin/src/main/java/org/apache/skywalking/oap/server/core/alarm/provider/welink/WeLinkSettings.java:56]: Yes，it should be 'robot'，I will fix it in my next commit.

File: oap-server/server-alarm-plugin/src/test/resources/alarm-settings.yml
Comment (YunfengGao) [reply oap-server/server-alarm-plugin/src/test/resources/alarm-settings.yml:128]: I can add some comment above this line and doc.

File: oap-server/server-alarm-plugin/src/main/java/org/apache/skywalking/oap/server/core/alarm/provider/Rules.java
Comment (YunfengGao) [reply oap-server/server-alarm-plugin/src/main/java/org/apache/skywalking/oap/server/core/alarm/provider/Rules.java:45]: I'll think about it after google some demo.

**Comment by wu-sheng** [comment]
@YunfengGao Ping me after you resolve all comments and tests passed locally

File: oap-server/server-alarm-plugin/src/test/resources/alarm-settings.yml
Comment (wu-sheng) [reply oap-server/server-alarm-plugin/src/test/resources/alarm-settings.yml:129]: My question is, are these IDs real? If so, this is a security leak. If not, please put other words on this.

File: oap-server/server-alarm-plugin/src/test/resources/alarm-settings.yml
Comment (wu-sheng) [inline oap-server/server-alarm-plugin/src/test/resources/alarm-settings.yml:130]:
```suggestion
      client_secret: dummy_secret_key
```
Like this.

Comment (YunfengGao) [reply oap-server/server-alarm-plugin/src/test/resources/alarm-settings.yml]: ok, but i wonder why cannot use "d7bc50ee-1995-0212-870f-64673cb797fe" here since it's not real.

Comment (wu-sheng) [reply oap-server/server-alarm-plugin/src/test/resources/alarm-settings.yml]: Because when you face end users, people don't know you, all documents could mislead. 
And using a clearly fake ID could have less risk.

File: docs/en/setup/backend/backend-alarm.md
Comment (wu-sheng) [inline docs/en/setup/backend/backend-alarm.md:268]: `client_id`, `group_ids` are still real numbers, please remove them.

Comment (YunfengGao) [reply docs/en/setup/backend/backend-alarm.md]: done.

**Review by wu-sheng — DISMISSED** [review]

**Comment by wu-sheng** [comment]
It seems you miss the update of changelog. Changes.md in the root.

**Review by wu-sheng — APPROVED** [review]

**Pull request merged**
